tests/telemetry/test_telemetry_metrics.py

```diff
@@ -0,0 +1,255 @@
+import sys
+
+from mock.mock import ANY
+import pytest
+
+from ddtrace.internal.constants import TELEMETRY_APPSEC
+from ddtrace.internal.constants import TELEMETRY_TRACER
+from ddtrace.internal.constants import TELEMETRY_TYPE_GENERATE_METRICS
+from ddtrace.internal.telemetry.metrics_namespaces import TelemetryTypeError
+from ddtrace.internal.utils.version import _pep440_to_semver
+from tests.telemetry.test_writer import _get_request_body
+
+
+def _assert_metric(test_agent_session_telemetry_metrics, expected_series, namespace=TELEMETRY_TRACER, seq_id=1):
+    test_agent_session_telemetry_metrics.telemetry_writer.periodic()
+    events = test_agent_session_telemetry_metrics.get_events()
+    # TODO: test_agent_session.clear() creates many telemetry events 'app-closing' type
+    assert len([event for event in events if event["request_type"] == TELEMETRY_TYPE_GENERATE_METRICS]) == 1
+
+    payload = {
+        "namespace": namespace,
+        "lib_language": "python",
+        "lib_version": _pep440_to_semver(),
+        "series": expected_series,
+    }
+    assert events[0]["request_type"] == TELEMETRY_TYPE_GENERATE_METRICS
+
+    assert events[0] == _get_request_body(payload, TELEMETRY_TYPE_GENERATE_METRICS, seq_id)
+
+
+@pytest.mark.skipif(sys.version_info < (3, 6), reason="mock.ANY doesn't works in py3.5 or lower")
+def test_send_metric_flush_and_series_is_restarted(test_agent_session_telemetry_metrics, mock_time):
+    """A datapoint is at least: a metric name, a metric value, and the time at which the value was collected.
```

Comment:
This docstring didn't help me understand the function. You might consider removing it here and in the few other places it's duplicated.